nomad/keyring_endpoint.go

```diff
@@ -112,16 +112,10 @@ func (k *Keyring) List(args *structs.KeyringListRootKeyMetaRequest, reply *struc
 
 	defer metrics.MeasureSince([]string{"nomad", "keyring", "list"}, time.Now())
 
-	// we need to allow both humans with management tokens and
-	// non-leader servers to list keys, in order to support
-	// replication
-	err := validateTLSCertificateLevel(k.srv, k.ctx, tlsCertificateLevelServer)
-	if err != nil {
-		if aclObj, err := k.srv.ResolveACL(args); err != nil {
-			return err
-		} else if aclObj != nil && !aclObj.IsManagement() {
-			return structs.ErrPermissionDenied
-		}
+	if aclObj, err := k.srv.ResolveACL(args); err != nil {
+		return err
+	} else if aclObj != nil && !aclObj.IsManagement() {
+		return structs.ErrPermissionDenied
 	}
```

Comment:
When servers call this endpoint do they pass an `aclObj`?

Reply:
As it turns out, servers don't call this `Keyring.List` endpoint at all! I think we originally had this to support keyring replication between leader and followers, but the followers just read the metadata from the state store and then `Keyring.Get` the actually key material.